atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java

```diff
@@ -294,7 +294,6 @@ public TimestampService getTimestampService() {
 
     public KeyValueServiceStatus getKeyValueServiceStatus() {
         NodeAvailabilityStatus nodeAvailabilityStatus = keyValueService.getNodeAvailabilityStatus();
-        // TODO: if config doesn't match db state return TERMINAL
```

Comment:
I think it's already done! the config/db check happens inside KVS.getNodeAvailabilityStatus.